### PR TITLE
[discussion] diff for linux

### PR DESCRIPTION
**Pull request by looooo**
dirty diff for linux. Some commits seem to be empty after rebasing...
Feel free to criticize...

### CODE REVIEW

**Comment by trelau** [comment]
Have you built this locally for Linux or any other platforms? It looks good (the builds are failing due to memory limits) but I did notice some of these types of warnings in Travis-CI: https://travis-ci.org/LaughlinResearch/pyOCCT/jobs/370770026#L1692-L1698

I think this is something that should be noted and fixed for the next iteration of the bindings, but just an FYI in case you try and use one of those types and get a hard crash. It seems like I screwed up the Deleter pattern for some types. Hopefully by the next iteration pybind11 accepts my PR where we won't need the Deleter template.

**Comment by looooo** [comment]
yes I am trying to get pyocct work with smesh and netgen. But so far I had not much success. But this is more related to SMESH: https://github.com/LaughlinResearch/SMESH/pull/4

**Comment by looooo** [comment]
I cleaned the commits. With the PR for SMESH it seems like the basic tests are working:
```
import OCCT
import OCCT.Exchange
import OCCT.Boolean
import OCCT.SMESH
```

File: src/SMESH.cpp [hunk 1]
```diff
@@ -1132,7 +1132,7 @@ PYBIND11_MODULE(SMESH, mod) {
 	// C:\Users\Trevor\Work\Products\SMESH\install\include\smesh\SMESH_MeshVSLink.hxx
 	py::class_<SMESH_MeshVSLink, opencascade::handle<SMESH_MeshVSLink>, MeshVS_DataSource3D> cls_SMESH_MeshVSLink(mod, "SMESH_MeshVSLink", "None");
 	cls_SMESH_MeshVSLink.def(py::init<const SMESH_Mesh *>(), py::arg("aMesh"));
-	cls_SMESH_MeshVSLink.def(py::init<const SMESH_subMesh *>(), py::arg("aSubMesh"));
+	// FIXME cls_SMESH_MeshVSLink.def(py::init<const SMESH_subMesh *>(), py::arg("aSubMesh"));
```
Comment (trelau) [inline src/SMESH.cpp]: Does this not work for you? I frequently use viewing of sub-meshes so I'd prefer to keep this in there, at least for windows. Can we put it behind a ``#if defined(WIN32)`` if we must? Or does this work after you updated some of the SMESH issues?

Comment (looooo) [reply src/SMESH.cpp]: I can't remeber the problem with this. I will try to include the line next time.
Now back to windows, it seems like I introduced a regression to smesh:
```
   Creating library bin\SMDS.lib and object bin\SMDS.exp
SMDS.cpp.obj : error LNK2001: unresolved external symbol "public: static int SMDS_Mesh::chunkSize" (?chunkSize@SMDS_Mesh@@2HA)
bin\SMDS.pyd : fatal error LNK1120: 1 unresolved externals
```

Comment (trelau) [reply src/SMESH.cpp]: For Windows, use ``ENABLE_FORCE=ON``. There are a few link errors I get where it's referencing static variables in SMESH. They are initialized in the header and defined in the source which is perfectly valid C++, plus for some reason is causing these errors in the pybind11 wrapping. I need figure out if it's the way I'm wrapping the class or something wrong with pybind11 that needs a fix. Either way, use that CMake option.

Comment (looooo) [reply src/SMESH.cpp]: I tried again and everything seems to compile without error (linux). PR is updated.

**Pull request merged**